nlmod/observations.py

```diff
@@ -0,0 +1,232 @@
+import numpy as np
+import xarray as xr
+from scipy.spatial import Delaunay
+
+from nlmod.dims import grid
+
+
+def compute_interpolation_weights(xy, uv, d=2):
+    """Calculate interpolation weights for barycentric interpolation [1]_.
```

Comment:
I think it is better to make this function and `interpolate_vertices_weights` private functions -> `_interpolate_vertices_weights` & `_compute_interpolation_weights`

Reply:
Good idea